src/Domain/TicTacToeGame.cs

```diff
@@ -0,0 +1,202 @@
+        {
+            InitializeBoard();
+            _moves = new List<Move>();
+            _currentPlayer = Player.Cross;
+            _winner = Winner.None;
+
+            ApplyMoves(moves);
+        }
+
+        public MoveResult Move(Move m)
+        {
+            try
+            {
+                ValidateNoWinner();
+                ValidateIsPlayersMove(m.Player);
+                ValidateValidBoardToPlay(m.BoardPosition);
+
+                SmallBoardInformation board = GetBoard(m.BoardPosition);
+                SmallTileInformation tile = GetTile(board, m.TilePosition);
+
+                ValidateTileEmpty(tile);
+
+                tile.Value = m.Player.ToTileValue();
+
+                Winner winner = board.Tiles.GetWinner(m.Player);
+                if (winner != Winner.None)
+                {
+                    board.Value = winner.ToTileValue();
+                    _winner = _board.GetWinner(m.Player);
+                }
+
+                // if there is no exception until here the move was valid
+                _moves.Add(m);
+                ChangeCurrentPlayer();
+
+                return new MoveResult
+                {
+                    IsValid = true,
+                    Move = m,
+                    MoveFinishedBoard = winner != Winner.None,
+                    MoveFinishedGame = _winner != Winner.None,
+                };
+            }
+            catch (InvalidMoveException e)
+            {
+                return new MoveResult
+                {
+                    IsValid = false,
+                    Move = m,
+                    MoveFinishedBoard = false,
+                    MoveFinishedGame = false,
+                    InvalidReason = e.Message,
+                };
+            }
+        }
+
+        private void ValidateNoWinner()
+        {
+            if (_winner != Winner.None)
+            {
+                throw new GameAlreadyFinishedException();
+            }
+        }
+
+        private void ValidateTileEmpty(SmallTileInformation tile)
+        {
+            if (tile.Value != TileValue.Empty)
+            {
+                throw new TileNotEmtpyException();
+            }
+        }
+
+        private void ValidateValidBoardToPlay(Position positionToValidate)
+        {
+            if (!_moves.Any())
+            {
+                //no moves ==> board is empty ==> all moves valid
+                return;
+            }
+
+            Move lastMove = _moves.Last();
+            Position lastMoveTilePosition = lastMove.TilePosition;
+
+            if (_board[lastMoveTilePosition.X][lastMoveTilePosition.Y].Value == TileValue.Empty)
+            {
+                if (lastMoveTilePosition.Equals(positionToValidate))
+                {
+                    return;
+                }
+
+                throw new IllegalPositionException();
+            }
+
+            if (_board[positionToValidate.X][positionToValidate.Y].Value == TileValue.Empty)
```

Comment:
I think you should put this into a well named variable because I thought several minutes about why we need this, which is a sign that the code is not 100% clear.

Reply:
any better name suggestions then nextBoardFromLastMove

Reply:
Maybe we should use a different approach.
What was not clear for me is that we need this `if` for the case that the last board was full. (maybe it was also a bit too late when I looked at it)